content/shmem_sync_all.tex

```diff
@@ -0,0 +1,44 @@
+\color{ForestGreen}
+\apisummary{
+    Registers the arrival of a \ac{PE} at a barrier and suspends \ac{PE}
+    execution until all other \acp{PE} arrive at the barrier.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_sync_all(void);
+\end{Csynopsis}
+
+\begin{apiarguments}
+
+    \apiargument{None.}{}{}
+
+\end{apiarguments}
+
+\apidescription{
+    The \FUNC{shmem\_sync\_all} routine registers the arrival of a \ac{PE} at a
+    barrier. Barriers are a fast mechanism for synchronizing all \acp{PE} at
+    once.  This routine causes a \ac{PE} to suspend execution until all
+    \acp{PE} have called \FUNC{shmem\_sync\_all}.
+
+    In contrast with the \FUNC{shmem\_barrier\_all} routine,
+    \FUNC{shmem\_sync\_all} only ensures completion of previously issued memory
+    stores and does not ensure completion of remote memory updates issued via
+    \openshmem routines.
+}
```

Comment:
The semantics here define the minimum completion guarantees that this function must provide.  It should be fine to generate additional completions.